nwbinspector/checks/nwbfile_metadata.py

```diff
@@ -114,17 +115,50 @@ def check_doi_publications(nwbfile: NWBFile):
 
 @register_check(importance=Importance.BEST_PRACTICE_SUGGESTION, neurodata_type=Subject)
 def check_subject_age(subject: Subject):
-    """Check if the Subject age is in ISO 8601."""
+    """Check if the Subject age is in ISO 8601 or our extension of it for ranges."""
     if subject.age is None:
         if subject.date_of_birth is None:
             return InspectorMessage(message="Subject is missing age and date_of_birth.")
-    elif not re.fullmatch(duration_regex, subject.age):
-        return InspectorMessage(
-            message=(
-                f"Subject age, '{subject.age}', does not follow ISO 8601 duration format, e.g. 'P2Y' for 2 years "
-                "or 'P23W' for 23 weeks."
-            )
+        else:
+            return
+    if re.fullmatch(pattern=duration_regex, string=subject.age):
+        return
+
+    if "/" in subject.age:
+        subject_lower_age_bound, subject_upper_age_bound = subject.age.split("/")
+
+        if re.fullmatch(pattern=duration_regex, string=subject_lower_age_bound) and (
+            re.fullmatch(pattern=duration_regex, string=subject_upper_age_bound) or subject_upper_age_bound == "**"
+        ):
+            return
+
+    return InspectorMessage(
+        message=(
+            f"Subject age, '{subject.age}', does not follow ISO 8601 duration format, e.g. 'P2Y' for 2 years "
+            "or 'P23W' for 23 weeks. You may also specify a range using a '/' separator, e.g., 'P1D/P3D' for an "
+            "age range somewhere from 1 to 3 days. If you cannot specify the upper bound of the range due to HIPAA "
+            "requirements, use '**' to leave it unspecified, e.g., 'P70Y/**' to mean an age greater than 70 years."
```

Comment:
```suggestion
            "age range somewhere from 1 to 3 days. If you cannot specify the upper bound of the range, "
            "you may leave the right side blank, e.g., 'P90Y/' means 90 years old or older."
```

Reply:
Even mentioning HIPPA is tricky because it might imply that we offer HIPPA compliance support. Let's just leave it out, but also give the exact use-case that HIPPA requirements would need.